client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -141,7 +141,12 @@ export default function getThankYouPageUrl( {
 		// extract a product from the cart, in siteless checkout there should only be one
 		const productSlug = cart?.products[ 0 ]?.product_slug;
 
-		return `/checkout/jetpack/thank-you/no-site/${ productSlug ?? 'no_product' }`;
+		const thankYouUrlSiteLess = `/checkout/jetpack/thank-you/no-site/${
+			productSlug ?? 'no_product'
+		}`;
+		return pendingOrReceiptId
+			? `${ thankYouUrlSiteLess }?receiptId=${ pendingOrReceiptId }`
```

Comment:
I'm sure there's a unit test for this URL... could you update it to cover the case where it gets this query param?

Reply:
Sorry for missing the obviously broken test. I just pushed a commit that makes the logic more robust and adds unit tests to cover it.

Reply:
🎉  thank you!